src/main/java/com/couchbase/lite/DatabaseUpgrade.java

```diff
@@ -167,11 +173,16 @@ protected void deleteSQLiteFiles() {
     }
 
     private boolean moveAttachmentsDir() {
+        // v1.1.x or v1.2.x
         File oldAttachmentsPath = new File(FileDirUtils.getPathWithoutExt(path) + " attachments");
-        if (!oldAttachmentsPath.canRead())
-            return true;
+        if (!oldAttachmentsPath.exists() || !oldAttachmentsPath.canRead()) {
+            // v1.0.x
+            oldAttachmentsPath = new File(FileDirUtils.getPathWithoutExt(path), "attachments");
+            if (!oldAttachmentsPath.exists() || !oldAttachmentsPath.canRead())
+                return false;
```

Comment:
In case of not existing the attachment directory, the method should return true, then go through. Thank you very much for catching this.